tests/functional/backend/corpora/test_api.py

```diff
@@ -229,20 +230,37 @@ def test_dataset_upload_flow(self):
         res.raise_for_status()
         self.assertEqual(res.status_code, requests.codes.ok)
         data = json.loads(res.content)
-        # TODO @madison update if we speed up backend
         self.assertEqual(data["upload_status"], "WAITING")
 
-        # Check non_auth user cant check status
-        no_auth_headers = {"Content-Type": "application/json"}
-        res = requests.get(f"{self.api}/dp/v1/datasets/{dataset_uuid}/status", headers=no_auth_headers)
-        with self.assertRaises(HTTPError):
-            res.raise_for_status()
+        with self.subTest("Test dataset conversion"):
+            keep_trying = True
+            upload_statuses = ["WAITING", "UPLOADING", "UPLOADED"]
+            conversion_statuses = ["CONVERTING", "CONVERTED", "FAILED"]
+            timer = time.time()
+            while keep_trying:
+                time.sleep(30)
+                res = requests.get(f"{self.api}/dp/v1/datasets/{dataset_uuid}/status", headers=headers)
+                res.raise_for_status()
+                self.assertIn(data["upload_status"], upload_statuses)
+                self.assertIn(data["conversion_cxg_status"], conversion_statuses)
+                if data["conversion_cxg_status"] == "FAILED":
+                    raise (f"CXG CONVERSION FAILED. Check logs for dataset: {dataset_uuid}")
```

Comment:
Use self.fail(f"CXG CONVERSION FAILED. Check logs for dataset: {dataset_uuid}"). https://docs.python.org/3/library/unittest.html#unittest.TestCase.fail